crypto/ml_kem/ml_kem.c

```diff
@@ -0,0 +1,1567 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Copyright (c) 2024, Google Inc. */
+
+#ifndef OPENSSL_NO_ML_KEM
+
+# include <assert.h>
+# include <internal/common.h>
+# include <internal/constant_time.h>
+# include <crypto/ml_kem.h>
+# include "ml_kem_local.h"
+
+typedef ossl_ml_kem_cbd_func cbd_t;
+typedef const ossl_ml_kem_vinfo *vinfo_t;
+
+# define DEGREE ML_KEM_DEGREE
+static const int kPrime = ML_KEM_PRIME;
```

Comment:
This notation of kX seems to come from boring.. I have not seen this used in OpenSSL before (although it seems to be in X25519 also - which probably also came from there)

Reply:
Yes, this is part of Google's internal C++ coding standards. Would like to not see it used here?

Reply:
OpenSSL seems to use _ and lowercase.